docs/language_guide.md

```diff
@@ -1174,130 +1280,653 @@ contract trait Ownable {
+    let extInstance = Ext(address: externalAddress)
+    do {
+      call extInstance.someFunction()
+    } catch is ExternalCallError {
+      // handle the error here
+    }
+  }
 }
+```
 
-// The functions in this block can be called by any user.
-Bank :: account <- (any) {
-  public init(manager: Address) {
-    self.manager = manager
+### Specifying hyper-parameters
+
+In addition to function parameters, there are two more "hyper-parameters" that need to be set when performing an external call.
+
+The `gas` hyper-parameter (defaults to `2300`) with type `Int` specifies how much Gas is allocated for the external call. Executing any code in EVM costs Gas and so the more Gas is provided, the more work can be done in a contract. The default amount, `2300`, is enough to emit a single event (at the time of writing).
+
+The `value` hyper-parameter (defaults to `0`) with type `Wei` specifies how much, if any, Wei is attached to the external call. Providing a non-zero amount causes money to be transferred from the calling (Flint) contract to the external contract. `value` must be specified if and only if calling a function marked as `@payable`.
+
+To specify `gas` and/or `value` for an external call, the syntax is:
+
+```swift
+call(<hyper-parameters>) <contract>.<function-name>(<parameters>)
+call(<hyper-parameters>)! <contract>.<function-name>(<parameters>)
+```
+
+Example:
+
+```swift
+X :: (any) {
+  public func callback(externalAddress: Address) {
+    let extInstance = Ext(address: externalAddress)
+    call(gas: 10000)! extInstance.someLongFunction()
+    call(value: Wei(unsafeRawValue: 100))! extInstance.someExpensiveFunction()
+    call(gas: 10000, value: Wei(unsafeRawValue: 100))! extInstance.someLongExpensiveFunction()
   }
+}
+```
+
+### Casting to and from Solidity types
+
+Since the types of external contract function parameters and return values are specified using [Solidity types](#solidity-types), values must be converted before they are used for an external call. This is facilitated using the type casting expression.
 
-  // Returns the manager's address.
-  public mutating func register() {
-    accounts[lastIndex] = account
-    lastIndex += 1
+Example:
+
+```swift
+X :: (any) {
+  public func callback(externalAddress: Address) {
+    let extInstance = Ext(address: externalAddress)
+    var flintInt: Int = 1
+    call! extInstance.someFunctionTakingAnInt(someParameter: flintInt as! int256)
+
+    flintInt = (call! extInstance.someReturningFunction()) as! Int
   }
+}
+```
+
+The forced cast (`as!`) expression converts Flint types to Solidity types and vice versa, after performing some basic runtime checks to make sure that the original value fits into the target value, since Solidity supports integer types of smaller sizes than the Flint default of 256 bits. An error results in the transaction being reverted.
+
+ > **Planned feature**
+ > 
+ > In the future, casting failures will be possible to handle using `do-catch` blocks.
```

Comment:
and with optionals (as?)

Reply:
Fun fact Swift doesn't support do-catch with casting.

as -> calls implicit constructor, this is a coercion and fails at compile time
as? -> does implicit constructor or reinterpret, optional nil on fail
as! -> reinterpret and crash if no conversion possible